bindings/src/main/java/io/pravega/storage/gcp/GCPChunkStorage.java

```diff
@@ -0,0 +1,243 @@
+
+    //region members
+    /**
+     * GCPStorageConfig contains config values.
+     */
+    private final GCPStorageConfig config;
+
+    /**
+     * Storage is provided by GCP to perform cloud operation on google cloud storage.
+     */
+    private final Storage storage;
+
+    //endregion
+
+    //region constructor
+    public GCPChunkStorage(Storage storage, GCPStorageConfig config, Executor executor, boolean shouldCloseClient) {
+        super(executor);
+        this.config = Preconditions.checkNotNull(config, "config");
+        this.storage = Preconditions.checkNotNull(storage, "client");
+    }
+    //endregion
+
+    //region capabilities
+
+    @Override
+    public boolean supportsConcat() {
+        return false;
+    }
+
+    @Override
+    public boolean supportsAppend() {
+        return false;
+    }
+
+    @Override
+    public boolean supportsTruncation() {
+        return false;
+    }
+
+    //endregion
+
+    //region implementation
+
+    @Override
+    protected ChunkHandle doOpenRead(String chunkName) throws ChunkStorageException {
+        if (!checkExists(chunkName)) {
+            throw new ChunkNotFoundException(chunkName, "doOpenRead");
+        }
+        return ChunkHandle.readHandle(chunkName);
+    }
+
+    @Override
+    protected ChunkHandle doOpenWrite(String chunkName) throws ChunkStorageException {
+        if (!checkExists(chunkName)) {
+            throw new ChunkNotFoundException(chunkName, "doOpenWrite");
+        }
+        return new ChunkHandle(chunkName, false);
+    }
+
+    @Override
+    protected int doRead(ChunkHandle handle, long fromOffset, int length, byte[] buffer, int bufferOffset) throws ChunkStorageException {
+        try (ReadChannel readChannel = this.storage.reader(this.config.getBucket(), getObjectPath(handle.getChunkName()))) {
+            try {
+                readChannel.seek(fromOffset);
+                readChannel.limit(fromOffset + length);
+                int val = readChannel.read(ByteBuffer.wrap(buffer, bufferOffset, length));
+                if (val == -1) {
+                    ChunkInfo info = doGetInfo(handle.getChunkName());
+                    if (fromOffset >= info.getLength() || (length + fromOffset) >= info.getLength()) {
+                        throw new IllegalArgumentException(handle.getChunkName());
+                    }
+                }
+                return val;
+            } catch (Exception e) {
+                throw convertException(handle.getChunkName(), "doRead", e);
+            }
+        }
+    }
+
+    @Override
+    protected int doWrite(ChunkHandle handle, long offset, int length, InputStream data) {
+        throw new UnsupportedOperationException("GCPChunkStorage does not support writing to already existing objects.");
+    }
+
+    @Override
+    public int doConcat(ConcatArgument[] chunks) throws UnsupportedOperationException {
+        throw new UnsupportedOperationException("GCP chunk storage does not support doConcat");
```

Comment:
This seems that is not covered by tests and seems easy to do so.

Reply:
Hi Raul, We have a story for concat feature. I will add as part of that.